### ISSUE TITLE
`entered unreachable code: Stored ENR is not contactable`

### ISSUE THREAD

**Issue by emhane** [sigp/discv5#243]
Goes through unreachable path.

https://github.com/sigp/discv5/blob/6707bcf3da350ff273208bb38dc29acc4cc4850c/src/service.rs#L598

Have a hunch this has to do with adding multiaddresses, since this is how I'm adding enodes in my pr, entry point here

https://github.com/paradigmxyz/reth/blob/993036014eb6c3cb51e308b885339e483771dbb4/crates/net/network/src/config.rs#L553

```
thread 'tokio-runtime-worker' panicked at /home/ubuntu/.cargo/git/checkouts/discv5-7af694118de5b208/04ac004/src/service.rs:598:29:                                                                                                                   
internal error: entered unreachable code: Stored ENR is not contactable. enr:-Le4QAAHJBTIJ1hP4eLJgvFFJ2E29tIVz9LEgQeACShg3sMxGgt-zi7bVlaHdEICENFuir0QAxMuW_Cj_39hhTjPUZSCB3-HYXR0bmV0c4hgAAAAAAAAAIRldGgykGmuDpkFAXAA__________-CaWSCdjSEcXVpY4IjM4lz
ZWNwMjU2azGhA7gc3n5LM3gKW91FEtG_4J_a-lgwJxot6tLQtHnmomEOiHN5bmNuZXRzDIN0Y3CCIzw                                                                                                                                                                      
stack backtrace:                                                                                                                                                                                                                                     
   0: rust_begin_unwind                                                                                                                                                                                                                              
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:645:5                                                                                                                                                   
   1: core::panicking::panic_fmt                                                                                                                                                                                                                     
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:72:14                                                                                                                                                  
   2: discv5::service::Service::handle_rpc_request                                                                                                                                                                                                   
             at /home/ubuntu/.cargo/git/checkouts/discv5-7af694118de5b208/04ac004/src/service.rs:598:29                                                                                                                                              
   3: discv5::service::Service::start::{{closure}}                                                                                                                                                                                                   
             at /home/ubuntu/.cargo/git/checkouts/discv5-7af694118de5b208/04ac004/src/service.rs:385:33                                                                                                                                              
   4: discv5::service::Service::spawn::{{closure}}::{{closure}}                                                                                                                                                                                      
             at /home/ubuntu/.cargo/git/checkouts/discv5-7af694118de5b208/04ac004/src/service.rs:328:33                                                                                                                                              
   5: <core::pin::Pin<P> as core::future::future::Future>::poll                                                                                                                                                                                      
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/future/future.rs:124:9                                                                                                                                              
   6: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}                                                                                                                                                                                       
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/core.rs:328:17                                                                                                                       
   7: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut                                                                                                                                                                                         
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/loom/std/unsafe_cell.rs:16:9                                                                                                                      
   8: tokio::runtime::task::core::Core<T,S>::poll                                                                                                                                                                                                    
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/core.rs:317:13                                                                                                                       
   9: tokio::runtime::task::harness::poll_future::{{closure}}                                                                                                                                                                                        
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:485:19                                                                                                                    
  10: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once                                                                                                                                                  
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panic/unwind_safe.rs:272:9                                                                                                                                          
  11: std::panicking::try::do_call                                                                                                                                                                                                                   
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:552:40                                                                                                                                                  
  12: __rust_try                                                                                                                                                                                                                                     
  13: std::panicking::try                                                                                                                                                                                                                            
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:516:19                                                                                                                                                  
  14: std::panic::catch_unwind                                                                                                                                                                                                                       
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panic.rs:142:14                                                                                                                                                      
  15: tokio::runtime::task::harness::poll_future                                                                                                                                                                                                     
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:473:18                                                                                                                    
  16: tokio::runtime::task::harness::Harness<T,S>::poll_inner                                                                                                                                                                                        
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:208:27                                                                                                                    
  17: tokio::runtime::task::harness::Harness<T,S>::poll                                                                                                                                                                                              
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:153:15                                                                                                                    
  18: tokio::runtime::task::raw::poll                                                                                                                                                                                                                
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/raw.rs:271:5                                                                                                                         
  19: tokio::runtime::task::raw::RawTask::poll                                                                                                                                                                                                       
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/raw.rs:201:18                                                                                                                        
  20: tokio::runtime::task::LocalNotified<S>::run                                                                                                                                                                                                    
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/mod.rs:416:9                                                                                                                         
  21: tokio::runtime::scheduler::multi_thread::worker::Context::run_task::{{closure}}                                                                                                                                                                
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:639:17                                                                                                   
  22: tokio::runtime::coop::with_budget                                                                                                                                                                                                              
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/coop.rs:107:5                                                                                                                             
  23: tokio::runtime::coop::budget                                                                                                                                                                                                                   
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/coop.rs:73:5                                                                                                                              
  24: tokio::runtime::scheduler::multi_thread::worker::Context::run_task                                                                                                                                                                             
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:575:9
  25: tokio::runtime::scheduler::multi_thread::worker::Context::run
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:526:24
  26: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:491:21
  27: tokio::runtime::context::scoped::Scoped<T>::set
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context/scoped.rs:40:9
  28: tokio::runtime::context::set_scheduler::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context.rs:176:26
  32: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:486:9
  33: tokio::runtime::context::runtime::enter_runtime
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/context/runtime.rs:65:16
  34: tokio::runtime::scheduler::multi_thread::worker::run
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:478:5
  35: tokio::runtime::scheduler::multi_thread::worker::Launch::launch::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/scheduler/multi_thread/worker.rs:447:45
  36: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/blocking/task.rs:42:21
  37: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/core.rs:328:17
  38: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/loom/std/unsafe_cell.rs:16:9
  39: tokio::runtime::task::core::Core<T,S>::poll
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/core.rs:317:13
  40: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:485:19
  41: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panic/unwind_safe.rs:272:9
  42: std::panicking::try::do_call
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:552:40
  43: __rust_try
  44: std::panicking::try
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:516:19
  45: std::panic::catch_unwind
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panic.rs:142:14
  46: tokio::runtime::task::harness::poll_future
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:473:18
  47: tokio::runtime::task::harness::Harness<T,S>::poll_inner 
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:208:27
  48: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/harness.rs:153:15
  49: tokio::runtime::task::raw::poll
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/raw.rs:271:5
  50: tokio::runtime::task::raw::RawTask::poll
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/raw.rs:201:18
  51: tokio::runtime::task::UnownedTask<S>::run
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/mod.rs:453:9
  52: tokio::runtime::blocking::pool::Task::run
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/blocking/pool.rs:159:9
  53: tokio::runtime::blocking::pool::Inner::run
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/blocking/pool.rs:513:17
  54: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/blocking/pool.rs:471:13
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by emhane** [comment]
related
```
2024-03-28T15:33:54.744117Z ERROR Trying to ping a non-contactable peer enr:-Li4QMwxIUunyM8OnJdeCei0ZtGqGBgjp6_BbIeO7G9iMv0HVelQTJWTG6FMYDY4a5EIRgJTI16MEamPSJ-3AuhuYguDAcBNh2F0dG5ldHOI__________-EZXRoMpBprg6ZBQFwAP__________gmlkgnY0hHF1aWOCIymJc2VjcDI1NmsxoQO1oWSybUvD1ZQwUL9SbRV2aKeIhvcC-I_UeT1OybpjeohzeW5jbmV0cw-DdGNwgiMo
```

**Comment by AgeManning** [comment]
Interesting, we'll look into it. 

If you're adding multiaddrs, not sure if these will help you or not:

We built an extension trait for ENR to add support for libp2p-related things: 
https://github.com/sigp/lighthouse/blob/stable/beacon_node/lighthouse_network/src/discovery/enr_ext.rs

Also, because I use this around in a few places, I made an effort to integrate it into Enr at one point, so you could just enable multiaddrs via a feature, but it got stalled:
https://github.com/sigp/enr/pull/60

**Comment by AgeManning** [comment]
Resolved in #246
